src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -15,30 +22,63 @@
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestParam;
 
+import javax.servlet.http.HttpServletRequest;
 import javax.validation.Valid;
+import java.util.Optional;
 
 @Controller
 @Api(tags = {"search"})
 @ConditionalOnProperty(name = "features.search.api", havingValue = "enabled")
 public class SearchApiController implements SearchApi {
 
+    private final ObjectMapper objectMapper;
+    private final HttpServletRequest request;
+    private final IamService iamService;
+    private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+    private final SearchService searchService;
+
     @Autowired
-    public SearchApiController() {
-        // do nothing
+    public SearchApiController(
+        ObjectMapper objectMapper,
+        HttpServletRequest request,
+        IamService iamService,
+        AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+        SearchService searchService
+    ) {
+        this.objectMapper = objectMapper;
+        this.request = request;
+        this.iamService = iamService;
+        this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+        this.searchService = searchService;
+    }
+
+    @Override
+    public Optional<ObjectMapper> getObjectMapper() {
+        return Optional.ofNullable(objectMapper);
```

Comment:
Not sure what TDR conventions are, but IMO `Optional.of()` would be appropriate here since `objectMapper` cannot be null.

Reply:
I copied this from the other controllers, but I can switch it.

Reply:
The way I think about it is, "how do I want the code to behave if x is null?" -- do we want it to fail early or fail late? Using `of` will fail early.